src/features/PowerShellNotebooks.ts

```diff
@@ -228,23 +291,36 @@ class PowerShellNotebookContentProvider implements vscode.NotebookContentProvide
         const retArr: string[] = [];
         for (const cell of document.cells) {
             if (cell.cellKind === vscode.CellKind.Code) {
-                retArr.push(...cell.document.getText().split(/\r|\n|\r\n/));
+                retArr.push(...cell.document.getText().split(/\r\n|\r|\n/));
             } else {
                 // First honor the comment type of the cell if it already has one.
                 // If not, use the user setting.
                 const commentKind = cell.metadata.custom?.commentType || Settings.load().notebooks.saveMarkdownCellsAs;
 
                 if (commentKind === CommentType.BlockComment) {
-                    retArr.push("<#");
-                    retArr.push(...cell.document.getText().split(/\r|\n|\r\n/));
-                    retArr.push("#>");
+                    const openBlockCommentOnOwnLine: boolean = cell.metadata.custom?.openBlockCommentOnOwnLine;
+                    const closeBlockCommentOnOwnLine: boolean = cell.metadata.custom?.closeBlockCommentOnOwnLine;
+                    const text = cell.document.getText().split(/\r\n|\r|\n/);
+                    if (openBlockCommentOnOwnLine) {
+                        retArr.push("<#");
+                    } else {
+                        text[0] = `<# ${text[0]}`;
+                    }
+
+                    if (!closeBlockCommentOnOwnLine) {
+                        text[text.length - 1] += " #>";
+                        retArr.push(...text);
+                    } else {
+                        retArr.push(...text);
+                        retArr.push("#>");
+                    }
                 } else {
-                    retArr.push(...cell.document.getText().split(/\r|\n|\r\n/).map((line) => `# ${line}`));
+                    retArr.push(...cell.document.getText().split(/\r\n|\r|\n/).map((line) => `# ${line}`));
                 }
             }
         }
 
-        await vscode.workspace.fs.writeFile(targetResource, new TextEncoder().encode(retArr.join("\n")));
+        await vscode.workspace.fs.writeFile(targetResource, new TextEncoder().encode(retArr.join(EOL)));
```

Comment:
This will replace newlines in a script when it changes operating systems. For example, if we have a script checked in to git and you edit it, then I edit it, I'll change all the newlines in it. Harder to implement an alternative, but something we should aim not to do -- I wouldn't want my text editor silently reencoding line endings as a side-effect of an unrelated operation. I think worth documenting and/or opening a new issue for

Reply:
So to solve this, I included metadata at the document level that says which line ending that file uses